### ISSUE TITLE
Automated build script

### ISSUE THREAD

**Issue by ZeyadYasser** [savoirfairelinux/opendht#235]
I was asking if the developers could provide an automated script for building OpenDHT on Ubuntu or prebuilt versions ..... I spent over 2 days trying to build using the build instructions on Ubuntu 14.04 LTS with no success.
Thanks

**Comment by AmarOk1412** [comment]
You already have the CMake, Autotools and if you want, the Dockerfile.

What did you tried?
What's the error?
Which gcc version do you use?

**Comment by ZeyadYasser** [comment]
I am on Ubuntu 14.04 LTS (gcc version 5.4.1) and the latest GnuTLS I can get through apt-get is "3.2.11" which is not suitable so I downloaded and built GnuTLS 3.3 successfully.
Now a new error raises when I build openDHT using "make -j4" cmd (following the build guide): 
`[  9%] Building CXX object CMakeFiles/opendht.dir/src/infohash.cpp.o
[ 11%] Building CXX object CMakeFiles/opendht-static.dir/src/crypto.cpp.o
[ 13%] Building CXX object CMakeFiles/opendht.dir/src/crypto.cpp.o
/home/zeyady/opendht/src/crypto.cpp: In member function ‘dht::InfoHash dht::crypto::PublicKey::getId() const’:
/home/zeyady/opendht/src/crypto.cpp:541:43: error: ‘GNUTLS_KEYID_USE_SHA256’ was not declared in this scope
     const int flags = (id.size() == 32) ? GNUTLS_KEYID_USE_SHA256 : 0;
                                           ^
/home/zeyady/opendht/src/crypto.cpp: In function ‘std::ostream& dht::crypto::operator<<(std::ostream&, const dht::crypto::TrustList::VerifyResult&)’:
/home/zeyady/opendht/src/crypto.cpp:1290:24: error: ‘GNUTLS_CERT_PURPOSE_MISMATCH’ was not declared in this scope
         if (h.result & GNUTLS_CERT_PURPOSE_MISMATCH)
                        ^
[ 15%] Building CXX object CMakeFiles/opendht-static.dir/src/default_types.cpp.o
make[2]: *** [CMakeFiles/opendht-static.dir/src/crypto.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 17%] Building CXX object CMakeFiles/opendht.dir/src/default_types.cpp.o
[ 19%] Building CXX object CMakeFiles/opendht.dir/src/node.cpp.o
/home/zeyady/opendht/src/crypto.cpp: In member function ‘dht::InfoHash dht::crypto::PublicKey::getId() const’:
/home/zeyady/opendht/src/crypto.cpp:541:43: error: ‘GNUTLS_KEYID_USE_SHA256’ was not declared in this scope
     const int flags = (id.size() == 32) ? GNUTLS_KEYID_USE_SHA256 : 0;
                                           ^
/home/zeyady/opendht/src/crypto.cpp: In function ‘std::ostream& dht::crypto::operator<<(std::ostream&, const dht::crypto::TrustList::VerifyResult&)’:
/home/zeyady/opendht/src/crypto.cpp:1290:24: error: ‘GNUTLS_CERT_PURPOSE_MISMATCH’ was not declared in this scope
         if (h.result & GNUTLS_CERT_PURPOSE_MISMATCH)
                        ^
make[2]: *** [CMakeFiles/opendht.dir/src/crypto.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/opendht-static.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/opendht.dir/all] Error 2
make: *** [all] Error 2
`

**Comment by aberaud** [comment]
Indeed build was broken on GnuTLS 3.3. It should now be fixed on master :-)

However GnuTLS 3.3 is going end of life and we recommend to upgrade to 16.04 LTS if possible.

I still hope to keep supporting OpenDHT on Ubuntu 14.04 for as long as possible. For Ring, we ended up dropping Ubuntu 14.04 because there was too many obsolete packages/tools (including obsolete GnuTLS) and we had to move on with our limited ressources.
